test/unit/storage-component.spec.ts

```diff
@@ -0,0 +1,202 @@
+import { AuthIdentity } from '@dcl/crypto'
+import { createStorageComponent } from '../../src/ports/storage/component'
+import { IStorageComponent } from '../../src/ports/storage/types'
+import { createTestIdentity, generateRandomIdentityId } from '../utils/test-identity'
+
+describe('when testing storage component', () => {
+  let storage: IStorageComponent
+  let validAuthIdentity: AuthIdentity
+  let identityId: string
+
+  beforeEach(async () => {
+    storage = createStorageComponent()
+
+    identityId = generateRandomIdentityId()
+
+    const testIdentity = await createTestIdentity()
+    validAuthIdentity = testIdentity.authChain
+  })
+
+  describe('when storing identity IDs', () => {
+    describe('and the identity ID is valid', () => {
+      let expiration: Date
+      let createdAt: Date
+
+      beforeEach(() => {
+        expiration = new Date(Date.now() + 60000)
+        createdAt = new Date()
+      })
+
+      it('should store the identity successfully', () => {
+        storage.setIdentity(identityId, {
+          identityId,
+          identity: validAuthIdentity,
+          expiration,
+          createdAt
+        })
+
+        const storedIdentity = storage.getIdentity(identityId)
+        expect(storedIdentity).not.toBeNull()
+        expect(storedIdentity?.identityId).toBe(identityId)
+        expect(storedIdentity?.identity).toEqual(validAuthIdentity)
+        expect(storedIdentity?.expiration).toEqual(expiration)
+        expect(storedIdentity?.createdAt).toEqual(createdAt)
+      })
+    })
+
+    describe('and the identity ID does not exist', () => {
+      let nonExistentId: string
+
+      beforeEach(() => {
+        nonExistentId = generateRandomIdentityId()
+      })
+
+      it('should return undefined when getting non-existent identity', () => {
+        const result = storage.getIdentity(nonExistentId)
+        expect(result).toBeNull()
+      })
+    })
+  })
+
+  describe('when deleting identity IDs', () => {
+    describe('and the identity ID exists', () => {
+      let expiration: Date
+      let createdAt: Date
+
+      beforeEach(() => {
+        expiration = new Date(Date.now() + 60000)
+        createdAt = new Date()
+      })
+
+      it('should delete the identity successfully', () => {
+        // First store the identity
+        storage.setIdentity(identityId, {
+          identityId,
+          identity: validAuthIdentity,
+          expiration,
+          createdAt
+        })
```

Comment:
Would you mind moving the storage of the identity to a `beforeEach` in the `and the identity ID exists` scope?